src/Request.php

```diff
@@ -228,6 +229,11 @@ public function getUri()
         if ($this->uri === null || is_string($this->uri)) {
             $this->uri = new HttpUri($this->uri);
         }
+
+        $headers = $this->getHeaders()->toArray();
+        if (array_key_exists('X-Forwarded-Proto', $headers) && $headers['X-Forwarded-Proto'] === 'https') {
```

Comment:
This will end up overriding scheme each time Uri is fetched from Request. Getters must not have side effects